sdk/cosmosdb/cosmos/src/common/RUConsumedManager.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import semaphore from "semaphore";
+
+/**
+ * @hidden
+ * @hidden
+ * Specifies Net RUConsumed
+ */
+export class RUConsumedManager {
+  private ruConsumed: number = 0;
+  private semaphore: any;
+
+  constructor() {
+    this.semaphore = semaphore(1);
+  }
+
+  getRUConsumed(): Promise<number> {
+    return new Promise((resolve) => {
+      this.semaphore.take(() => {
+        this.semaphore.leave();
+        resolve(this.ruConsumed);
+      });
+    });
+  }
+
+  addToRUConsumed(value: number): Promise<void> {
+    return new Promise((resolve) => {
+      this.semaphore.take(() => {
```

Comment:
Why is a semaphore needed for synchronous operations? I thought JavaScript has run-to-completion semantics but maybe I missed something?